### DISCUSSION TITLE
Add typescript support to provided single page application setup

### DISCUSSION THREAD

**Discussion by mario-fehr** [sulu/SuluHeadlessBundle#106] [Ideas]
I am not sure how many people use the provided single page application setup from the headless bundle, but what do you think about providing typscript types/interfaces for these files out of the box?

**Comment by alexander-schranz** [comment]
@mfehr94 The js files in the Sulu Headless Bundle must more seen as an example, or a quickstart to experimental with the headless bundle. They are deeply connected to react and mobx and so they only represent how we develop our frontend application. I'm not sure how we continue with the JS implementation, if we refractored them to be more framework independent or have them as own skeleton. As it is more common when people using the headless bundle that they use other technologies like Next.js or Nuxt.js where they have Server Side rendering. So at current state there is not planned to add here typescript types.
